core/contract/evm/state.go

```diff
@@ -0,0 +1,153 @@
+package evm
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/hyperledger/burrow/acm"
+	"github.com/hyperledger/burrow/binary"
+	"github.com/hyperledger/burrow/crypto"
+	"github.com/hyperledger/burrow/permission"
+
+	"github.com/xuperchain/xuperchain/core/contract/bridge"
+)
+
+type stateManager struct {
+	ctx *bridge.Context
+}
+
+func newStateManager(ctx *bridge.Context) *stateManager {
+	return &stateManager{
+		ctx: ctx,
+	}
+}
+
+// Get an account by its address return nil if it does not exist (which should not be an error)
+func (s *stateManager) GetAccount(address crypto.Address) (*acm.Account, error) {
+	addr, addrType, err := DetermineEVMAddress(address)
+	if err != nil {
+		return nil, nil
+	}
+
+	var evmCode []byte
+	if addrType == contractNameType {
+		v, err := s.ctx.Cache.Get("contract", evmCodeKey(addr))
+		if err != nil {
+			return nil, nil
+		}
+		evmCode = v.GetPureData().GetValue()
+	}
+
+	balance, err := s.ctx.Core.GetBalance(addr)
+	if err != nil {
+		return nil, nil
+	}
+	return &acm.Account{
+		Address:     address,
+		Balance:     balance,
+		EVMCode:     evmCode,
+		Permissions: permission.AllAccountPermissions,
+	}, nil
+}
+
+// Retrieve a 32-byte value stored at key for the account at address, return Zero256 if key does not exist but
+// error if address does not
+func (s *stateManager) GetStorage(address crypto.Address, key binary.Word256) ([]byte, error) {
+	fmt.Printf("\nget %s %s\n", address, key)
+	fmt.Printf("first address %s\n", s.ctx.ContractName)
+	contractName, _, err := DetermineEVMAddress(address)
```

Comment:
This function will be called frequently, can the address here be asserted as a contract address